### DISCUSSION TITLE
Error on computing accessible name for a heading element

### DISCUSSION THREAD

**Discussion by juanca** [eps1lon/dom-accessibility-api#581] [bug] [answered]
Let me know if this is better issued somewhere else. 

---

node@14.15.1
jest@26.6.3
@testing-library/react@9.5.0
jsdom@16.4.0 from jest#@jest#core#jest-config#jest-environment-jsdom

---

I have a test that waits for a combobox to appear. While it waits, this library (+ JSDOM) throws `Not implemented: window.computedStyle(elt, pseudoElt)`. I've tried to set up a codesanbox but I think running the tests in the browser is doing something weird... 

Here is the stack trace (truncated):

![image](https://user-images.githubusercontent.com/841084/112765432-0e395d80-8fc2-11eb-8d0b-0fd2d5da8e45.png)

The function call that shouldn't (?) happen:

![image](https://user-images.githubusercontent.com/841084/112765472-35902a80-8fc2-11eb-9877-070a838fcce2.png)

The node in question:

![image](https://user-images.githubusercontent.com/841084/112765484-4476dd00-8fc2-11eb-9f7a-582fb806237c.png)

**Comment by eps1lon** [comment, marked as answer]
You want to use `computeAccessibleName(element, { computedStyleSupportsPseudoElements: false })` in `JSDOM`. Note that this will not return accurate results but the only workaround available to us. See https://github.com/jsdom/jsdom/issues/3025 for more context.

**Reply by juanca** [reply]
Sound good. Thanks for the advise. I will go up the stack trace and report the issue in the corresponding repository.
